husarion_ugv_gazebo/launch/spawn_charging_docks.launch.py

```
@@ -0,0 +1,125 @@
+from launch.conditions import IfCondition
+from launch.substitutions import LaunchConfiguration, PathJoinSubstitution
+from launch_ros.actions import Node
+from launch_ros.substitutions import FindPackageShare
+
+
+def generate_apriltag_and_get_path(tag_id):
+    from moms_apriltag import TagGenerator2
+
+    tag_generator = TagGenerator2("tag36h11")
+    tag_image = tag_generator.generate(tag_id, scale=1000)
+    temp_file = NamedTemporaryFile(suffix=f"_tag_{tag_id}.png", delete=False)
```

Ruff (0.8.0)

Use a context manager for opening files

(SIM115)